File: backend/src/main/kotlin/com/example/backend/models/ProduitDetail.kt
package com.example.backend.models

import jakarta.persistence.*
import org.hibernate.annotations.ColumnDefault

@Entity
@Table(name = "produit_detail")
class ProduitDetail {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY) // Use IDENTITY for auto-increment
  @Column(name = "id", nullable = false)
  var id: Int? = null

  @Column(name = "reference", length = 32)
  var reference: String? = null

  @Column(name = "nom", nullable = false, length = 50)
  var nom: String? = null

  @Column(name = "stock", nullable = false)
  var stock: Int? = null

  @Column(name = "stockMax", nullable = false)
  var stockMax: Int? = null

  @Column(name = "stockMin", nullable = false)
  var stockMin: Int? = null

  @ColumnDefault("0")
  @Column(name = "reductionMax", nullable = false)
  var reductionMax: Int? = null

  @Column(name = "prix", nullable = false)
  var prix: Int? = 0

  @Column(name = "grossiste_list", nullable = false, length = 100)
  var grossisteList: String? = null

  @ColumnDefault("0")
  @Column(name = "supprimer", nullable = false)
  var  supprimer: Int? = 0
}

File: backend_py/pharmaPython/app/models/produit_detail.py
from sqlalchemy import Column, Integer, String
from app.api.v1.db import Base

class ProduitDetail(Base):
  __tablename__ = "produit_detail"
  __table_args__ = {"extend_existing": True}

  id = Column(Integer, primary_key=True, autoincrement=True)
  reference = Column(String(32))
  nom = Column(String(50), nullable=False)
  stock = Column(Integer, nullable=False)
  stock_max = Column(Integer, nullable=False)
  stock_min = Column(Integer, nullable=False)
  reduction_max = Column(Integer, default=0, nullable=False)
  prix = Column(Integer, nullable=False)
  grossiste_list = Column(String(100), nullable=False)
  supprimer = Column(Integer, default=0, nullable=False)
